### PR TITLE
Bragg plot bugs fixed.

### PR DESCRIPTION
**Pull request by Kvieta1990**
Original bug identified by Antonio - when the checked bank (to plot) does not exist in the loaded file, ADDIE will crash.

> Fix: We first check whether the selected bank to plot exists in all loaded files. If yes, we go ahead to plot. If not, error message will be printed out to terminal without killing ADDIE.

Another bug identified by myself during testing and debugging the first bug mentioned above - in `Multiple Banks` mode, if multiple datasets exist, sometimes it can happen that if no data set is selected, unchecking a bank box will lead to ADDIE crash.

> Fix: When multiple data sets exist in the workspace and none of them is selected, checking and unchecking a certain box will lead to re-plotting of relevant banks for all data sets.

### CODE REVIEW

File: addie/rietveld/braggtree.py [hunk 1]
```diff
@@ -236,6 +236,34 @@ def do_plot_ws(self):
...
+                for key, list_temp in leaf_dict_temp.items():
+                    for item_temp_1 in list_temp:
+                        if item_temp in item_temp_1:
+                            bank_temp = list_temp.index(item_temp_1) + 1
+                            self._main_window._braggBankWidgets[bank_temp].setChecked(True)
+                            parent_found = True
+                            parent_in_tree = key
+                            break
+                    if parent_found:
+                        break
```
Comment (marshallmcdonnell) [inline addie/rietveld/braggtree.py:255]: Looks like this is repeated below in lines 350-366 (correct me if wrong)
From DRY (Don't Repeat Yourself) principle, maybe we should move this to a function / class method and reuse that in both spots.
Thoughts?

File: addie/rietveld/braggtree.py [hunk 2]
```diff
@@ -317,11 +345,39 @@ def do_select_gss_node(self):
...
```
Comment (marshallmcdonnell) [inline addie/rietveld/braggtree.py:366]: Looks like this is repeated below in lines 239-255 (correct me if wrong)
See comment above at those lines for suggestion.

**Review by marshallmcdonnell — CHANGES_REQUESTED** [review]
Thanks for the fixes!
Overall, looks good, just two comments / suggestions:
 - Move some of the duplicate code to a function (see comments at relevant lines of code)
 - Small but do you mind to remove the "ZYP" part from the comments in code? We can track that info using `git blame` and can become stale with future changes. Great comments though, please leave them!

File: addie/rietveld/braggtree.py (hunk 1 above)
Comment (Kvieta1990) [reply addie/rietveld/braggtree.py:255]: > Looks like this is repeated below in lines 350-366 (correct me if wrong)
> From DRY (Don't Repeat Yourself) principle, maybe we should move this to a function / class method and reuse that in both spots.
> Thoughts?

Yes. Indeed there is a duplicate piece of these codes here. Will create a function.

**Comment by Kvieta1990** [comment]
> Thanks for the fixes!
> Overall, looks good, just two comments / suggestions:
> 
> * Move some of the duplicate code to a function (see comments at relevant lines of code)
> * Small but do you mind to remove the "ZYP" part from the comments in code? We can track that info using `git blame` and can become stale with future changes. Great comments though, please leave them!

Just committed changes to reflect your comments. Thank you!

**Review by marshallmcdonnell — APPROVED** [review]
LGTM!

**Pull request merged**